scripts/main/contextMenu.js

```diff
@@ -79,6 +79,13 @@ contextMenu.add = function (e) {
 					});
 				}
 			}
+			if (!album.isSmartID(albumID) && lychee.map_display) {
+				// display track add button if it's a regular album
+				items.push({}, { title: build.iconic("location") + lychee.locale["UPLOAD_TRACK"], fn: () => $("#upload_track_file").click() });
```

Comment:
Why does the event handler triggers another click-event? Why don't you directly call the relevant method `upload.uploadTrack`?

Reply:
Before calling the method, a file chooser mist be shown (this is the same behavior as the image upload is doing it).

Reply:
You are right. I don't like the frontend but that is nothing you could be blamed for. So we keep it that way for consistency reasons, although I hate it. I regard this conversation as resolved.